datalake/src/main/java/com/sequenceiq/datalake/service/sdx/DistroxService.java

```diff
@@ -31,4 +51,66 @@
             throw e;
         }
     }
+
+    public void stopAttachedDistrox(String envName) {
+        Collection<StackViewV4Response> attachedDistroXClusters = getAttachedDistroXClusters(envName, null);
+        List<String> pollingCrn = new ArrayList<>();
+        attachedDistroXClusters.forEach(d -> {
+            if (!d.getStatus().isStopState()) {
+                distroXV1Endpoint.putStopByCrn(d.getCrn());
```

Comment:
If any of these rest calls fail for any reason, the whole stop operation will fail. There could be temporary network issues or anything.

There are 2 ways to mitigate this:
- better: introduce multi-stop and multi-start endpoints
- worse: handle various exceptions here